app/scripts/modules/core/src/presentation/modal/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+
+const { useMemo } = React;
+
+import { useEventListener, useContainerClassNames, useLatestCallback } from 'core/presentation';
+
+import { ModalContext } from './ModalContext';
+import * as styles from './Modal.module.css';
+
+export interface IModalProps {
+  isOpen: boolean;
+  onRequestClose?: () => any;
+  children?: React.ReactNode;
+}
+
+export const Modal = ({ onRequestClose, isOpen, children }: IModalProps) => {
+  useContainerClassNames(isOpen ? [styles.backdropBlurEffect] : []);
+
+  const keydownCallback = ({ keyCode }: KeyboardEvent) => {
+    if (keyCode === 27) {
+      onRequestClose();
+    }
+  };
+  useEventListener(document, 'keydown', isOpen ? keydownCallback : null);
```

Comment:
really slick 👍

Reply:
while it's not _super_ common to need this hook, it's amazing to use when you do need it. Totally different world from the class lifecycle way of doing it.